File: motomanager/frontend/motomanager-web/src/app/models/material.ts
export interface Material {
  id: number;
  name: string;
  unitOfMeasureId: number;
  unitOfMeasureName?: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMaterialRequest {
  name: string;
  unitOfMeasureId: number;
}

export interface UpdateMaterialRequest {
  name: string;
  unitOfMeasureId: number;
  isActive: boolean;
}

export interface UnitOfMeasure {
  id: number;
  name: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateUnitOfMeasureRequest {
  name: string;
}

export interface UpdateUnitOfMeasureRequest {
  name: string;
  isActive: boolean;
}

export interface ServiceOperation {
  id: number;
  name: string;
  workHours: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateServiceOperationRequest {
  name: string;
  workHours: number;
}

export interface UpdateServiceOperationRequest {
  name: string;
  workHours: number;
  isActive: boolean;
}

File: motomanager/backend/MotoManager.Domain/Entities/Material.cs
namespace MotoManager.Domain.Entities;

public class Material
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public long UnitOfMeasureId { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public UnitOfMeasure? UnitOfMeasure { get; set; }
}
